org.lflang.diagram/src/org/lflang/diagram/lsp/LFLanguageServer.java

```diff
@@ -18,17 +20,19 @@ public class LFLanguageServer extends KGraphLanguageServerExtension {
     public void cancelProgress(WorkDoneProgressCancelParams params) {
         Progress.cancel(params.getToken().getRight().intValue());
     }
+    boolean initialized = false;
 
     @Override
     protected Hover hover(HoverParams params, CancelIndicator cancelIndicator) {
         // This override is just a hacky little patch that is being applied downstream of the original mistake and
         //  upstream of the ungraceful handling (IndexOutOfBoundsException) of said mistake. This patch is applied here
         //  simply because it is easy. This would be done differently were it not for the fact that we plan to rebuild
         //  this infrastructure from scratch anyway.
-        try {
-            return super.hover(params, cancelIndicator);
-        } catch (IndexOutOfBoundsException e) {
-            return IHoverService.EMPTY_HOVER;  // Fail silently
+        if (!initialized) {
+            CppLanguageServer.init();
+            initialized = true;
         }
+        return CppLanguageServer.hoverRequest(params);
+        // return IHoverService.EMPTY_HOVER;  // Fail silently
```

Comment:
```suggestion
        try {
            if (!initialized) {
                CppLanguageServer.init();
                initialized = true;
            }
            return CppLanguageServer.hoverRequest(params);
        } catch (IndexOutOfBoundsException e) {
            return IHoverService.EMPTY_HOVER;  // Fail silently
        }
```
The old code which got deleted in this PR does "Fail silently," as it says, which is of course very bad. However, I want to keep it around until we move away from Xtext. See https://github.com/lf-lang/vscode-lingua-franca/issues/63 (and the comment about the "hacky little patch") for background

Reply:
Oh, wait, never mind. If we don't use the default hover method, we don't have the `IndexOutOfBoundsException` from the framework, so that comment was not exactly right. But we do use the default hover method if we are not hovering over C++ code. So actually, the right thing to do is to use the old hacky thing only if we are not hovering over C++.